--- __init__.py.orig	2010-04-15 01:23:43.000000000 +0200
+++ __init__.py	2010-08-13 15:17:39.011410249 +0200
@@ -754,44 +754,30 @@
     def doBuildinstall(self):
         """Run anaconda-runtime's buildinstall on the tree."""
 
+        # initialize yum object
+        self._inityum()
 
-        # setup the buildinstall call
-        buildinstall = ['/usr/lib/anaconda-runtime/buildinstall']
-        #buildinstall.append('TMPDIR=%s' % self.workdir) # TMPDIR broken in buildinstall
+        # get the required stuff for lorax
+        yb = self.ayum
+        installtree = self.ayum._conf.installroot
+        outputdir = self.topdir
+
+        product = self.config.get('pungi', 'name')
+        version = self.config.get('pungi', 'version')
+        release = "%s %s" % (product, version)
+
+        workdir = self.workdir
+
+        variant = self.config.get('pungi', 'flavor')
+        bugurl = self.config.get('pungi', 'bugurl')
+
+        # run lorax
+        import pylorax
+        lorax = pylorax.Lorax(yb, installtree, outputdir,
+                              product, version, release,
+                              workdir, variant, bugurl)
 
-        buildinstall.append('--product')
-        buildinstall.append(self.config.get('pungi', 'name'))
-
-        if not self.config.get('pungi', 'flavor') == "":
-            buildinstall.append('--variant')
-            buildinstall.append(self.config.get('pungi', 'flavor'))
-
-        buildinstall.append('--version')
-        buildinstall.append(self.config.get('pungi', 'version'))
-
-        buildinstall.append('--release')
-        buildinstall.append('%s %s' % (self.config.get('pungi', 'name'), self.config.get('pungi', 'version')))
-
-        if self.config.has_option('pungi', 'bugurl'):
-            buildinstall.append('--bugurl')
-            buildinstall.append(self.config.get('pungi', 'bugurl'))
-
-        buildinstall.append('--output')
-        buildinstall.append(self.topdir)
-
-        for mirrorlist in self.mirrorlists:
-            buildinstall.append('--mirrorlist')
-            buildinstall.append(mirrorlist)
-
-        buildinstall.append(self.topdir)
-
-        # Add any extra repos of baseurl type
-        for repo in self.repos:
-            buildinstall.append(repo)
-
-        # run the command
-        # TMPDIR is still broken with buildinstall.
-        pypungi.util._doRunCommand(buildinstall, self.logger) #, env={"TMPDIR": self.workdir})
+        lorax.run()
 
         # write out the tree data for snake
         self.writeinfo('tree: %s' % self.mkrelative(self.topdir))
