diff -Naur a/share/efi.tmpl b/share/efi.tmpl
--- a/share/efi.tmpl	2012-01-12 16:05:52.000000000 +0100
+++ b/share/efi.tmpl	2012-01-13 12:23:47.238239411 +0100
@@ -10,7 +10,6 @@
 
 ## actually make the EFI images
 ${make_efiboot("images/efiboot.img")}
-${make_efiboot("images/macboot.img", imgtype="apple")}
 
 ## This is kinda gross, but then... so's EFI.
 <%def name="make_efiboot(img, include_kernel=False, disk=False, imgtype='default')">
diff -Naur a/share/x86.tmpl b/share/x86.tmpl
--- a/share/x86.tmpl	2012-01-12 16:05:52.000000000 +0100
+++ b/share/x86.tmpl	2012-01-13 14:10:04.168384841 +0100
@@ -53,10 +53,10 @@
     <%
     efiarch = 'X64' if basearch=='x86_64' else 'IA32'
     efigraft="EFI/BOOT={0}/EFI/BOOT".format(outroot)
-    for img in ("images/efiboot.img", "images/macboot.img"):
+    for img in ("images/efiboot.img", ):
         efiargs += " -eltorito-alt-boot -e {0} -no-emul-boot".format(img)
         efigraft += " {0}={1}/{0}".format(img,outroot)
-    efihybrid="--uefi --mac"
+    efihybrid="--uefi"
     %>
     <%include file="efi.tmpl" args="configdir=configdir, KERNELDIR=KERNELDIR, efiarch=efiarch"/>
 %endif
